pkg/fqdn/dnsproxy/proxy_test.go

```diff
@@ -524,62 +540,286 @@ func (s *DNSProxyTestSuite) TestFullPathDependence(c *C) {
+			restore.RuleRegex{Regexp: s.proxy.allowed[epID3][53][cachedDstID4Selector]}: map[string]struct{}{},
+		},
+	}
+	restored3 := s.proxy.GetRules(uint16(epID3))
+	c.Assert(restored3, checker.DeepEquals, expected3)
+
+	s.proxy.UpdateAllowed(epID1, 53, nil)
+	s.proxy.UpdateAllowed(epID1, 54, nil)
+	_, exists := s.proxy.allowed[epID1]
+	c.Assert(exists, Equals, false)
+
+	_, exists = s.proxy.allowed[epID2]
+	c.Assert(exists, Equals, false)
+
+	s.proxy.UpdateAllowed(epID3, 53, nil)
+	_, exists = s.proxy.allowed[epID3]
+	c.Assert(exists, Equals, false)
+
+	dstIP1 := (s.dnsServer.Listener.Addr()).(*net.TCPAddr).IP
+	dstIP2a := net.ParseIP("127.0.0.1")
+	dstIP2b := net.ParseIP("127.0.0.2")
+	dstIPrandom := net.ParseIP("127.0.0.42")
+
+	// Before restore: everything is dropped
+	// Case 1 | EPID1 | DstID1 |   53 | www.ubuntu.com | Rejected | No rules
+	allowed, err = s.proxy.CheckAllowed(epID1, 53, dstID1, dstIP1, "www.ubuntu.com")
+	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
+	c.Assert(allowed, Equals, false, Commentf("request was allowed when it should be rejected"))
+
+	// Case 2 | EPID1 | DstID1 |   54 | cilium.io      | Rejected | No rules
+	allowed, err = s.proxy.CheckAllowed(epID1, 54, dstID1, dstIP1, "cilium.io")
+	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
+	c.Assert(allowed, Equals, false, Commentf("request was allowed when it should be rejected"))
+
+	// Case 3 | EPID1 | DstID2 |   53 | cilium.io      | Rejected | No rules
+	allowed, err = s.proxy.CheckAllowed(epID1, 53, dstID2, dstIP2a, "cilium.io")
+	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
+	c.Assert(allowed, Equals, false, Commentf("request was allowed when it should be rejected"))
+
+	// Case 4 | EPID1 | DstID2 |   53 | aws.amazon.com | Rejected | No rules
+	allowed, err = s.proxy.CheckAllowed(epID1, 53, dstID2, dstIP2b, "aws.amazon.com")
+	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
+	c.Assert(allowed, Equals, false, Commentf("request was allowed when it should be rejected"))
+
+	// Case 5 | EPID1 | DstID1 |   54 | example.com    | Rejected | No rules
+	allowed, err = s.proxy.CheckAllowed(epID1, 54, dstID1, dstIP1, "example.com")
+	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
+	c.Assert(allowed, Equals, false, Commentf("request was allowed when it should be rejected"))
+
+	// Restore rules
+	s.proxy.RestoreRules(uint16(epID1), restored1)
+	_, exists = s.proxy.restored[epID1]
+	c.Assert(exists, Equals, true)
+
+	// Same tests with 2 (WORLD) dstID to make sure it is not used, but with correct destionation IP
+
+	// Case 1 | EPID1 | dstIP1 |   53 | www.ubuntu.com | Allowed
+	allowed, err = s.proxy.CheckAllowed(epID1, 53, 2, dstIP1, "www.ubuntu.com")
+	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
+	c.Assert(allowed, Equals, true, Commentf("request was rejected when it should be allowed"))
```

Comment:
nit: The comments in this section could benefit from mentioning that the allow is expected from the "restore of DNS rules", just to differentiate from earlier failures.

Reply:
Missed this sorry, will address in the next round if needed.

Reply:
Done.